### PR TITLE
Implemented advanced merging logic for VCF records

### PR DESCRIPTION
**Pull request by mhsaul**
Implemented merging logic to handle non variants and mnps in merge_with_non_variants_strategy.

### CODE REVIEW

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 1]
```diff
@@ -60,63 +61,140 @@ def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
...
+
+  def _merge_variants(self, grouped_variants):
+    merged_variants = []
+    for merge_key, variants in grouped_variants.iteritems():
+      merged_variants.extend(
+          self._move_to_calls.get_merged_variants(variants, merge_key))
+    return merged_variants
+
+  def _update_splits(self, splits, v):
+    # Increase split size to also combine adjacent splits
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: nit: please put this comment just above overlapping_splits.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 2]
```diff
@@ -60,63 +61,140 @@ def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
...
+      return
+    start = self._get_window_start(key)
+    end = start + self._window_size
+    if v.start < start:
+      v.start = start
+    if v.end > end:
+      v.end = end
+
+  def _is_non_variant(self, v):
+    return not v.reference_bases or v.start + len(v.reference_bases) < v.end
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Please check "not v.alternate_bases" or "<*>" or "<NON_REF>". I noticed that the <*> notation is actually from vcf4.3 (section 5.5: http://samtools.github.io/hts-specs/VCFv4.3.pdf)

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 3]
```diff
@@ -150,9 +150,13 @@ class MergeOptions(VariantTransformsOptions):
   # - NONE: Variants will not be merged across files.
   # - MOVE_TO_CALLS: uses libs.variant_merge.move_to_calls_strategy
   #   for merging. Please see the documentation in that file for details.
+  # - MERGE_WITH_NON_VARIANTS: uses
+  #   libs.variant_merge.merge_with_non_varaitns_strategy for merging. Please
+  #   see the documentation in that file for details.
   VARIANT_MERGE_STRATEGIES = [
       NONE,
       MOVE_TO_CALLS,
+      MERGE_WITH_NON_VARIANTS,
```
Comment (arostamianfar) [inline gcp_variant_transforms/options/variant_transform_options.py]: Please remove for now as discussed.

File: gcp_variant_transforms/third_party/__init__.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2017 Google Inc.  All Rights Reserved.
```
Comment (arostamianfar) [inline gcp_variant_transforms/third_party/__init__.py]: Please remove for now as discussed.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py (hunk 1 above)
Comment (mhsaul) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Whoops, looks like I moved some other stuff around and forgot this comment. Good catch!

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py (hunk 2 above)
Comment (mhsaul) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Done.

File: gcp_variant_transforms/options/variant_transform_options.py (hunk 3 above)
Comment (mhsaul) [reply gcp_variant_transforms/options/variant_transform_options.py]: Done.

File: gcp_variant_transforms/third_party/__init__.py (hunk 4 above)
Comment (mhsaul) [reply gcp_variant_transforms/third_party/__init__.py]: I removed the whole third party library from the history and replaced it with the pypi package.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 5]
```diff
@@ -60,63 +63,145 @@ def __init__(self, info_keys_to_move_to_calls_regex, copy_quality_to_calls,
...
+    end = start + self._window_size
+    if v.start < start:
+      v.start = start
+    if v.end > end:
+      v.end = end
+
+  def _is_non_variant(self, v):
+    if not v.alternate_bases:
+      return True
+    for alternate in NON_VARIANT_ALTERNATES:
```
Comment (arostamianfar) [inline gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: nit: you can write
return any(v.alternate_bases == [a] for a in NON_VARIANT_ALTERNATES)

Comment (mhsaul) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Good call! Done.

File: gcp_variant_transforms/options/variant_transform_options.py [hunk 6]
```diff
@@ -150,6 +150,9 @@ class MergeOptions(VariantTransformsOptions):
   # - NONE: Variants will not be merged across files.
   # - MOVE_TO_CALLS: uses libs.variant_merge.move_to_calls_strategy
   #   for merging. Please see the documentation in that file for details.
+  # - MERGE_WITH_NON_VARIANTS: uses
```
Comment (arostamianfar) [inline gcp_variant_transforms/options/variant_transform_options.py]: nit: remove this as well.

**Pull request merged**